arangod/Scheduler/Scheduler.cpp

```diff
@@ -254,25 +306,30 @@ bool Scheduler::queue(RequestPriority prio,
 
   switch (prio) {
     case RequestPriority::HIGH:
-      if (0 < _fifoSize[0]) {
-        ok = pushToFifo(static_cast<int>(prio), callback);
+      if (0 < _fifoSize[FIFO1]) {
+        ok = pushToFifo(FIFO1, callback, false);
       } else if (canPostDirectly()) {
-        post(callback);
+        post(callback, false);
       } else {
-        ok = pushToFifo(static_cast<int>(prio), callback);
+        ok = pushToFifo(FIFO1, callback, false);
       }
       break;
     case RequestPriority::LOW:
-      if (0 < _fifoSize[0]) {
-        ok = pushToFifo(static_cast<int>(prio), callback);
-      } else if (0 < _fifoSize[1]) {
-        ok = pushToFifo(static_cast<int>(prio), callback);
+      if (0 < _fifoSize[FIFO1]) {
+        ok = pushToFifo(FIFO2, callback, false);
+      } else if (0 < _fifoSize[FIFO8]) {
+        ok = pushToFifo(FIFO2, callback, false);
+      } else if (0 < _fifoSize[FIFO2]) {
+        ok = pushToFifo(FIFO2, callback, false);
       } else if (canPostDirectly()) {
-        post(callback);
+        post(callback, false);
       } else {
-        pushToFifo(static_cast<int>(prio), callback);
+        ok = pushToFifo(FIFO2, callback, false);
       }
```

Comment:
All this is not understandable for me.
the initial `if` checks `FIFO1` and pushes to `FIFO2`. The next `else if` checks `FIFO8` and pushes to `FIFO2`, then next `else if` checks `FIFO2` and pushes to `FIFO2`. And the final `else` pushes to `FIFO2`. I am lost.
Maybe this is all correct, but without any code comment I can't understand the intention behind it.